Assets/Scripts/Transformation.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Transformation : MonoBehaviour {
+
+	[SerializeField] private float transDuration = 1.2f, humanFormDuration = 5f;
+
+	private SpriteRenderer mySpriteRend;
+
+	private TransformationStates transStates;
+
+	private Color humanColor = Color.white, wolfColor = Color.black, transformingColor = Color.grey;
+
+	private float transTimer, humanFormTimer;
+
+
+	private void Start() {
+		mySpriteRend = gameObject.GetComponent<SpriteRenderer>();
+	}
+
+
+	private void Update() {
+		if (Input.GetKeyDown(KeyCode.R) && transStates == TransformationStates.Wolf) {
+			StartCoroutine(CoTransforming());
+		}
+		else if (Input.GetKeyDown(KeyCode.R) && transStates != TransformationStates.Wolf) {
+		}
+
+		if (transStates == TransformationStates.Human) {
+			mySpriteRend.color = humanColor;
+		}
+		else if (transStates == TransformationStates.Transforming) {
+			mySpriteRend.color = transformingColor;
+		}
+		else if (transStates == TransformationStates.Wolf) {
+			mySpriteRend.color = wolfColor;
+		}
+	}
+
+
+	private IEnumerator CoTransforming() {
+		transStates = TransformationStates.Transforming;
```

Comment:
What about having this as a boolean instead? Would be more generic if you could see if transforming and keep it from a middle. Might depend on if the human to wolf transformation should be instant.